### ISSUE TITLE
NVDAHelper Windows 10, switch from WRL/CXX to C++/WinRT

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#10662]
We are using Windows Runtime C++ Template Library (WRL) to in operate with Windows OneCore APIs, particularly the OCR and ONeCore APIs.

From the [WRL Docs](https://docs.microsoft.com/en-gb/windows/uwp/cpp-and-winrt-apis/index):

> WRL is now superseded by C++/WinRT, a standard C++17 language projection for Windows Runtime APIs. C++/WinRT is available in the Windows 10 SDK from version 1803 onward. C++/WinRT is implemented entirely in header files, and designed to provide you with first-class access to the modern Windows API.
With C++/WinRT, you can both consume and author Windows Runtime APIs using any standards-compliant C++17 compiler. C++/WinRT typically performs better and produces smaller binaries than any other language option for the Windows Runtime. We will continue to support C++/CX and WRL, but highly recommend that new applications use C++/WinRT.

While NVDA works with WRL just fine, we should investigate a switch to WinRT as soon as new features have to be added to the Windows 10 local lib.

**Comment by leonardder** [comment]
Blocked by #9661

**Comment by codeofdusk** [comment]
#9661 has been merged, so this issue is no longer blocked.

**Comment by feerrenrut** [comment]
Further investigation will need to be done into the implications of this work.

**Comment by leonardder** [comment]
Resources relevant for this process:

* [Move to C++/WinRT from C++/CX - UWP applications](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/move-to-winrt-from-cx)
* [Asynchrony, and interop between C++/WinRT and C++/CX](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/interop-winrt-cx-async)
* [Move to C++/WinRT from WRL](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/move-to-winrt-from-wrl)
* [String handling](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/strings)

We also need to find out how to write directly to a buffer (utils.cpp)
